src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -59,7 +60,24 @@ protected override void ExecuteCore()
             {
                 Directory.CreateDirectory(destinationDirectory);
             }
-            File.WriteAllBytes(ModifiedAppHostPath, array);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Why do we need this "if" check. The "else" section would work for Windows as well and keep the implementation consistent.